provider/aws.go

```diff
@@ -271,6 +281,37 @@ func (p *AWSProvider) records(zones map[string]*route53.HostedZone) ([]*endpoint
 				ep := endpoint.
 					NewEndpointWithTTL(wildcardUnescape(aws.StringValue(r.Name)), endpoint.RecordTypeCNAME, ttl, aws.StringValue(r.AliasTarget.DNSName)).
 					WithProviderSpecific(providerSpecificEvaluateTargetHealth, fmt.Sprintf("%t", aws.BoolValue(r.AliasTarget.EvaluateTargetHealth)))
+				newEndpoints = append(newEndpoints, ep)
+			}
+
+			for _, ep := range newEndpoints {
+				if r.SetIdentifier != nil {
+					ep.SetIdentifier = aws.StringValue(r.SetIdentifier)
+					if r.Weight != nil {
```

Comment:
maybe it makes sense to use switch case here, WDYT?

Reply:
Will do.